code/game/objects/structures/curtains.dm

```diff
@@ -33,11 +138,20 @@
 
 /obj/structure/curtain/proc/toggle()
 	set_opacity(!opacity)
+	cut_overlays()
 	if(opacity)
 		icon_state = "closed"
+		overlay = image("closed_overlay")
+		overlay.color = overlay_color
+		overlay.alpha = overlay_alpha
+		add_overlay(overlay)
 		layer = SHOWER_CLOSED_LAYER
 	else
 		icon_state = "open"
+		overlay = image("open_overlay")
+		overlay.color = overlay_color
+		overlay.alpha = overlay_alpha
+		add_overlay(overlay)
 		layer = SHOWER_OPEN_LAYER
```

Comment:
Could you move these to an `update_overlays()` call (and similar procs for the other things) and use the `update_appearance()` framework? I see this is duplicated